### Homework 3 Statistical Inference

#### Rania Siddiqui 07494
#### rs07494@st.habib.edu.pk

In [1]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
from scipy import stats

In [2]:
df = pd.read_csv('student-scores.csv')

In [3]:
df.head()

id first_name last_name                                  email  gender  \
0   1       Paul     Casey         paul.casey.1@gslingacademy.com    male   
1   2   Danielle  Sandoval  danielle.sandoval.2@gslingacademy.com  female   
2   3       Tina   Andrews       tina.andrews.3@gslingacademy.com  female   
3   4       Tara     Clark         tara.clark.4@gslingacademy.com  female   
4   5    Anthony    Campos     anthony.campos.5@gslingacademy.com    male   

   part_time_job  absence_days  extracurricular_activities  \
0          False             3                       False   
1          False             2                       False   
2          False             9                        True   
3          False             5                       False   
4          False             5                       False   

   weekly_self_study_hours   career_aspiration  math_score  history_score  \
0                       27              Lawyer          73             81   
1                       47              Doctor          90             86   
2                       13  Government Officer          81             97   
3                        3              Artist          71             74   
4                       10             Unknown          84             77   

   physics_score  chemistry_score  biology_score  english_score  \
0             93               97             63             80   
1             96              100             90             88   
2             95               96             65             77   
3             88               80             89             63   
4             65               65             80             74   

   geography_score  
0               87  
1               90  
2               94  
3               86  
4               76

### Q1

H0 -> math_score of males and females are same

Ha -> Math scores of males is greater than females

In [9]:
male_score = df[(df['gender'] == 'male')]['math_score']

female_score = df[(df['gender'] == 'female')]['math_score']

stats.ttest_ind(male_score, female_score, alternative='greater')

TtestResult(statistic=2.2371559026936256, pvalue=0.012693020299630226, df=1998.0)

Since, p_value = 0.012 < 0.05 we can reject null hypothesis that is math scores of males and females are the same. And so we have to accept the alternative hypothesis that says math scores of males is greater than females.

### Q2

HO -> math scores of non doctors is same as that of doctors

Ha -> math scores of non doctors is greater than doctors

In [14]:
math_score_non_doctors = df[(df['career_aspiration'] != 'Doctor')]['math_score']
math_score_doctors = df[(df['career_aspiration'] == 'Doctor')]['math_score']

stats.ttest_ind(math_score_non_doctors,math_score_doctors, alternative='greater')

TtestResult(statistic=-7.389083205876682, pvalue=0.999999999999892, df=1998.0)

Since, p_value = 0.99 > 0.5, we accept the null hypothesis and reject the alternative hypothesis which says that math score of non-doctors is greater than that of doctors. So we come to a conclusion that math scores of doctors and non doctors are same.

### Q3

HO -> weekly_self_study_hours of males and females are same

H1 -> weekly_self_study_hours of males are greater than females.

In [19]:
male_weekly_hours = df[(df['gender'] == 'male')]['weekly_self_study_hours']

female_weekly_hours = df[(df['gender'] == 'female')]['weekly_self_study_hours']

stats.ttest_ind(male_weekly_hours, female_weekly_hours, alternative='greater')

TtestResult(statistic=3.4567831650474443, pvalue=0.00027905018418441824, df=1998.0)

Since, p_value = 0.0027 < 0.05, we can reject the null hypothesis which says that weekly self study hours of males and females are the same. And so for that reason we have to accept the alternative hypothesis which says weekly self study hours of males are greater than females.

### Q4

Ho -> math score of no part time male students is equal to or same as part time male students

Ha -> math score of no part time male students is greater than part time male students

In [60]:
no_part_time_male = df[(df['part_time_job'] == False) & (df['gender'] == 'male')]['math_score']

part_time_male = df[(df['part_time_job'] == True) & (df['gender'] == 'male')]['math_score']
stats.ttest_ind(no_part_time_male, part_time_male, alternative='greater')

TtestResult(statistic=5.8026672393780085, pvalue=4.3820239750124745e-09, df=996.0)

Since p_value < 0.05, we will reject the null hypothesis and accept the alternative hypothesis which says that males who dont work part time, their math scores are greater than those who work part time. This shows that having a job alongside studies gets a little hard to manage as indicated by the math scores. 

### Q5

HO -> Average of history score is same as geography score

Ha -> Average of history score is less than geography score

In [31]:
stats.ttest_ind(df['history_score'], df['geography_score'] , alternative="less")

TtestResult(statistic=-1.4412586066465385, pvalue=0.07479497633658547, df=3998.0)

Since p_value 0.074 > 0.05, we accept the null hypothesis and conclude that average of history score is same as geography score. This also means that there is insufficient evidence to show that the avergae of history score is same as geography score.

### Q6

HO -> Average of math score is same as biology score

Ha -> Average of math score is greater than biology score

In [36]:
stats.ttest_ind(df['math_score'], df['biology_score'] , alternative="greater")

TtestResult(statistic=9.082620143235266, pvalue=8.146293509285797e-20, df=3998.0)

Since, p_value is less than 0.05, we reject the null hypothesis and accept the alternative hypothesis and conclude that average of math score is greater than biology score. 

### Q7

Ho -> weekly_self_study_hours of extracurricular_activities is same as self study hours of no 
extracurricular_activities

Ha -> weekly_self_study_hours of extracurricular_activities is less than no
extracurricular_activities..

In [41]:
hours_extra_curricular_activities = df[(df['extracurricular_activities'] == True)]['weekly_self_study_hours']

no_hours_extra_curricular_activities = df[(df['extracurricular_activities'] == False)]['weekly_self_study_hours']

stats.ttest_ind(hours_extra_curricular_activities, no_hours_extra_curricular_activities, alternative="less")

TtestResult(statistic=-2.3506159912870577, pvalue=0.009419553161020007, df=1998.0)

Since p_value is less than 0.05, we reject the null hypothesis and accept the alternative hypothesis. So we can conclude that people who are engaged in extra curricular activities have less time for self study. 

### Q8

HO -> absence days of females are same as males

Ha -> absence days of females are less than males

In [46]:
absence_days_fem = df[(df['gender'] == 'female')]['absence_days']

absence_days_male = df[(df['gender'] == 'male')]['absence_days']

stats.ttest_ind(absence_days_fem, absence_days_male, alternative='less')

TtestResult(statistic=1.3128102331149203, pvalue=0.9053011661354035, df=1998.0)

Since, p_value = 0.905 > 0.05, we can accept the null hypothesis and say that absence days of females are same as males and reject the alternative hypothesis.

### Q9

Ho =  There is no association between gender and career_aspiration.

Ha = There is an association between gender and career_aspiration.

In [51]:
chisqt = pd.crosstab(df['gender'],df['career_aspiration'], margins=True)
stats.chi2_contingency(chisqt)

Chi2ContingencyResult(statistic=155.39466648000175, pvalue=1.9087221746002932e-17, dof=34, expected_freq=array([[  63.126,   33.567,   84.669,  154.809,   34.068,   28.056,
          59.619,   31.563,   30.561,   69.138,   41.583,   19.539,
         157.815,   36.573,   29.559,  111.723,   16.032, 1002.   ],
       [  62.874,   33.433,   84.331,  154.191,   33.932,   27.944,
          59.381,   31.437,   30.439,   68.862,   41.417,   19.461,
         157.185,   36.427,   29.441,  111.277,   15.968,  998.   ],
       [ 126.   ,   67.   ,  169.   ,  309.   ,   68.   ,   56.   ,
         119.   ,   63.   ,   61.   ,  138.   ,   83.   ,   39.   ,
         315.   ,   73.   ,   59.   ,  223.   ,   32.   , 2000.   ]]))

Since, p_value = 1.90x10^-17 < 0.05, we reject the null hypothesis and accept the alternative hypothesis. So we conclude that there is an association between gender and career aspiration. 

### Q10

Ho = There is no association between gender and part time job.

Ha = There is an association between gender and part time jobn.

In [56]:
chisqt = pd.crosstab(df['gender'],df['part_time_job'], margins=True)
stats.chi2_contingency(chisqt)

Chi2ContingencyResult(statistic=1.1337105392319442, pvalue=0.8888874344046543, dof=4, expected_freq=array([[ 843.684,  158.316, 1002.   ],
       [ 840.316,  157.684,  998.   ],
       [1684.   ,  316.   , 2000.   ]]))

Since p_value = 0.88 > 0.05, we accept the null hypothesis and conclude that there is no association between gender and part time job. 